src/impl/tpls/KokkosBlas1_scal_tpl_spec_decl.hpp

```diff
@@ -460,4 +460,204 @@ KOKKOSBLAS1_CSCAL_TPL_SPEC_DECL_CUBLAS(Kokkos::LayoutLeft, Kokkos::CudaUVMSpace,
 
 #endif
 
+// rocBLAS
+#ifdef KOKKOSKERNELS_ENABLE_TPL_ROCBLAS
+#include <KokkosBlas_tpl_spec.hpp>
+
+namespace KokkosBlas {
+namespace Impl {
+
+#define KOKKOSBLAS1_DSCAL_TPL_SPEC_DECL_ROCBLAS(LAYOUT, MEMSPACE,              \
+                                                ETI_SPEC_AVAIL)                \
+  template <class ExecSpace>                                                   \
+  struct Scal<                                                                 \
+      Kokkos::View<double*, LAYOUT, Kokkos::Device<ExecSpace, MEMSPACE>,       \
+                   Kokkos::MemoryTraits<Kokkos::Unmanaged> >,                  \
+      double,                                                                  \
+      Kokkos::View<const double*, LAYOUT, Kokkos::Device<ExecSpace, MEMSPACE>, \
+                   Kokkos::MemoryTraits<Kokkos::Unmanaged> >,                  \
+      1, true, ETI_SPEC_AVAIL> {                                               \
+    typedef Kokkos::View<double*, LAYOUT, Kokkos::Device<ExecSpace, MEMSPACE>, \
+                         Kokkos::MemoryTraits<Kokkos::Unmanaged> >             \
+        RV;                                                                    \
+    typedef double AV;                                                         \
+    typedef Kokkos::View<const double*, LAYOUT,                                \
+                         Kokkos::Device<ExecSpace, MEMSPACE>,                  \
+                         Kokkos::MemoryTraits<Kokkos::Unmanaged> >             \
+        XV;                                                                    \
+    typedef typename XV::size_type size_type;                                  \
+                                                                               \
+    static void scal(const RV& R, const double& alpha, const XV& X) {          \
+      Kokkos::Profiling::pushRegion("KokkosBlas::scal[TPL_ROCBLAS,double]");   \
+      const size_type numElems = X.extent(0);                                  \
+      if ((numElems < static_cast<size_type>(INT_MAX)) &&                      \
+          (R.data() == X.data())) {                                            \
+        scal_print_specialization<RV, AV, XV>();                               \
+        const int N       = static_cast<int>(numElems);                        \
+        constexpr int one = 1;                                                 \
+        KokkosBlas::Impl::RocBlasSingleton& s =                                \
+            KokkosBlas::Impl::RocBlasSingleton::singleton();                   \
+        KOKKOS_ROCBLAS_SAFE_CALL_IMPL(                                         \
+            rocblas_dscal(s.handle, N, &alpha, R.data(), one));                \
```

Comment:
Check pointer mode?

Reply:
@e10harvey Do you mean for alpha? Scaling parameters like alpha and beta used in functions like axpy, gemv, gemm, scal can be allocated either on host memory or on device memory. But I think KK interfaces does not support scaling parameters on device yet. If there is any use case, we can revisit this for these functions above in another PR. For now, I think it is okay for not checking pointer mode here.